omc3/scripts/bad_bpms_summary.py

```diff
@@ -0,0 +1,294 @@
+    return df
+
+
+def read_iforest_bad_bpms_file(iforest_file: Path) -> tfs.TfsDataFrame:
+    """ Reads an iforest bad-bpm file and returns a TfsDataFrame with all unique bad-bpms.
+    
+    Args:
+        iforest_file (Path): Path to the bad-bpm file.
+
+    Returns:
+        tfs.TfsDataFrame: TfsDataFrame with all unique bad-bpms.
+
+    """
+    df_iforest = tfs.read(iforest_file)
+    plane = iforest_file.stem[-1]
+
+    df = get_empty_df()
+    df.loc[:, NAME] = list(set(df_iforest[NAME]))  # hint: be sure to ignore index
+    df.loc[:, PLANE] = plane.upper()
+    df.loc[:, SOURCE] = IFOREST 
+    df.loc[:, FILE] = str(iforest_file)
+    return df
+
+
+# Evaluaion ----
+
+
+def evaluate(df: tfs.TfsDataFrame) -> tfs.TfsDataFrame:
+    """ Evaluates the gathered bad-bpms and returns a TfsDataFrame with the results.
+
+    The evaluation is based on the following criteria:
+    - Count how often a BPM is bad
+    - Count the total number of (unique) files for each combination of accelerator, source and plane
+
+    From this information the percentage of how often a BPM is deemed bad is calculated.
+    
+    Args:
+        df (tfs.TfsDataFrame): TfsDataFrame with all bad-bpms.
+
+    Returns:
+        tfs.TfsDataFrame: TfsDataFrame with the evaluated results.
+    """
+    # Count how often a BPM is bad
+    df_counted = df.groupby([NAME, ACCEL, SOURCE, PLANE]).size().reset_index(name=COUNT)
+
+    # Count the total number of (unique) files for each combination of accelerator, source and plane
+    file_count = df.groupby([ACCEL, SOURCE, PLANE])[FILE].nunique().reset_index(name=FILE_COUNT)
+    df_counted = df_counted.merge(file_count, how="left", on=[ACCEL, SOURCE, PLANE])
+
+    df_counted.loc[:, PERCENTAGE] = round(
+        (df_counted[COUNT] / df_counted[FILE_COUNT]) * 100, 2
+    )
+    
+    df_counted = tfs.TfsDataFrame(df_counted.sort_values(PERCENTAGE, ascending=False), headers=df.headers)
+    return df_counted
+
+
+def print_results(df_counted: tfs.TfsDataFrame, print_percentage: float):
+    """ Prints the results to the console (INFO level). 
+    
+    Args:
+        df_counted (tfs.TfsDataFrame): TfsDataFrame with the evaluated results.
+        print_percentage (float): Print out BPMs that appear in more than this percentage of measurements.
+    """
+    percentage_mask = df_counted[PERCENTAGE] >= print_percentage
+
+    for accel in sorted(df_counted[ACCEL].unique()):
+        accel_mask = df_counted[ACCEL] == accel
+        for source in sorted(df_counted[SOURCE].unique()):
+            source_mask = df_counted[SOURCE] == source
+            df_filtered = df_counted.loc[percentage_mask & source_mask & accel_mask, :]
+            msg = "\n".join(
+                f"{row[NAME]:>20s} {row[PLANE]}: {row[PERCENTAGE]:5.1f}% ({row[COUNT]}/{row[FILE_COUNT]})" 
+                for _,row in df_filtered.iterrows()
+            )
+            LOG.info(f"Highest bad BPMs of {accel} from {source}:\n" + msg)
```

Comment:
Does this print to console always? If not, is this what the user would expect from the doc?

Reply:
The logging setup is done in `get_logger(__name__)`. So if you run this as a script, this is always printed. If you import the function and have no logger set up, then not. I will adapt